packages/data-point/lib/entity-types/entity-request/resolve.js

```diff
@@ -48,28 +25,58 @@ module.exports.getRequestOptions = getRequestOptions
 
 /**
  * Resolve options object
- * @param {Accumulator} acc
- * @param {function} resolveReducer
+ * @param {Accumulator} accumulator
+ * @param {Function} resolveReducer
+ * @return {Promise<Accumulator>}
  */
-function resolveOptions (acc, resolveReducer) {
-  const options = acc.reducer.spec.options
-  const transformOptionKeys = acc.reducer.spec.transformOptionKeys
-  // iterate over each option transform key
-  return Promise.reduce(
-    transformOptionKeys,
-    (newOptions, key) => {
-      return resolveReducer(acc, key.transform).then(res => {
-        return fp.set(key.path, res.value, newOptions)
-      })
-    },
-    options
-  ).then(resolvedOptions => {
-    return utils.set(acc, 'options', resolvedOptions)
+function resolveOptions (accumulator, resolveReducer) {
+  const options = accumulator.reducer.spec.options
+  return resolveReducer(accumulator, options).then(acc => {
+    const resolvedOptions = getRequestOptions(acc.value)
+    if (resolvedOptions.baseUrl && !resolvedOptions.uri) {
```

Comment:
this conditional block was previously in the `resolveBeforeRequest` function, but I tried to consolidate the code for generating `options`